### DISCUSSION TITLE
Async Storage Wagmi in Browser Extension

### DISCUSSION THREAD

**Discussion by jqphu** [wevm/wagmi#1630] [Q&A]
Hey!

I'm trying to get wagmi to run in a browser extension.

Everything works fine but I'm having a little trouble with the storage. I want to use `browser.storage.local` but this returns an async API - https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/StorageArea/get.

Using the default local storage doesn't seem recommended as it isn't shared in the background scripts.

https://developer.chrome.com/docs/extensions/reference/storage/

> Even though extensions can access [Window.localStorage](https://developer.mozilla.org/docs/Web/API/Window/localStorage) in some contexts (popup and other HTML pages), it is not recommended for the following reasons:
> The extension's service worker cannot access Window.localStorage.
> Content scripts share the Window.localStorage of the host page.
> Data saved to Window.localStorage is lost when the user clears their browsing history.

Any help is greatly appreciated - thanks :D

**Comment by iketw** [comment]
Any updates on this issue? I'm having the same problem when trying to use wagmi in react-native
